tests/conftest.py

```diff
@@ -307,6 +316,30 @@ def logout_validator(validator_index, key):
     return logout_validator
 
 
+@pytest.fixture
+def mk_withdraw(casper_ct, casper_address):
+    def mk_withdraw(validator_index, casper_chain, key):
+        print('key: ' + utils.encode_hex(key))
+        withdraw_func = casper_ct.encode('withdraw', [validator_index])
+        withdraw_tx = Transaction(
+            casper_chain.head_state.get_nonce(utils.privtoaddr(key)),
+            GAS_PRICE,
+            500000,
+            casper_address,
+            0,
+            withdraw_func
+        ).sign(key)
+        return withdraw_tx
+    return mk_withdraw
+
+@pytest.fixture
+def withdraw_validator(mk_withdraw):
+    def withdraw_validator(validator_index, casper_chain, key):
+        withdraw_tx = mk_withdraw(validator_index, casper_chain, key)
+        casper_chain.direct_tx(withdraw_tx)
```

Comment:
Why not just a `casper.withdraw(validator_index)`? 
Are you trying to avoid the use of the ABIContract?

Reply:
I was stuck why withdraw is failed (before found that contract cannot send because it cannot pay fees) and made this to be sure everything is done right. Will try `casper.withdraw` back instead